spec/unit/pops/lookup/context_spec.rb

```diff
@@ -171,13 +171,8 @@ module Lookup
 
       context 'and multiple compilations' do
 
-        before(:each) { Puppet.settings[:environment_timeout] = 'unlimited' }
-        after(:each) do
-          Puppet.settings[:environment_timeout] = 0
-          Puppet.lookup(:environments).clear_all
-        end
-
         it 'will reuse cached_file_data and not call block again' do
+          Puppet.settings[:environment_timeout] = 'unlimited'
```

Comment:
Please note that the test executed OK when the `environment_timeout` was set to "unlimited" before each test in the context.
Setting `environment_timeout` to 0 effectively destroys the assertion that the test is trying to make. With that setting, the cache is cleared between each compile so of course the test passes. The file is always loaded! The test is there to assert that the cache is refreshed due to a file change, not due to environment recreation. If that doesn't work, then the code that is tested is broken. The test executed OK before though, so I don't think that's the case.